resources/resolver.go

```diff
@@ -861,24 +908,47 @@ func resolveConfigAndTokenizerConfig(resources *Resources, hfConfig *HFConfig) (
 				if padToken, ok := config.(map[string]interface{})["pad_token"].(string); ok {
 					hfConfig.PadTokenStr = &padToken
 				}
-				hasReadConfig = true
+				hasReadForEOSBOS = true
 			}
-		}
-		if tokenizerConfig != nil && !hasReadConfig {
-			// Using interfaces, first check if bos_token is in string format
-			if bosToken, ok := tokenizerConfig.(map[string]interface{})["bos_token"].(string); ok {
-				hfConfig.BosTokenStr = &bosToken
-				if eosToken, ok := tokenizerConfig.(map[string]interface{})["eos_token"].(string); ok {
-					hfConfig.EosTokenStr = &eosToken
-				}
-				if padToken, ok := tokenizerConfig.(map[string]interface{})["pad_token"].(string); ok {
-					hfConfig.PadTokenStr = &padToken
+
+			// Read for EOS BOS token ID
+			if eosTokenId, ok := config.(map[string]interface{})["eos_token_id"].(float64); ok {
```

Comment:
Rather than doing `config.(map[string]interface{})` repeatedly, assign that to a variable and use that variable.